### PR TITLE
First implementation on autonomous trigger

### PR DESCRIPTION
**Pull request by AurelienFT**
* [x] document all added functions
* [x] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

### CODE REVIEW

File: massa-execution-worker/src/tests/scenarios_mandatories.rs
Comment (Eitu33) [inline massa-execution-worker/src/tests/scenarios_mandatories.rs:342]: This could use an extra inline comment to mark the line that triggers the SC. Also I believe you left some dead code commented in the test.

File: massa-final-state/src/final_state.rs
Comment (Eitu33) [inline massa-final-state/src/final_state.rs:248]: This is way too verbose for nothing, please use:

```suggestion
                            .filter_map(|change| match change {
                                Change::Add(id, _) | Change::Activate(id) | Change::Delete(id)
                                    if id <= &last_id =>
                                {
                                    Some(change.clone())
                                }
                                _ => None,
                            })
```

**Review by Eitu33 — COMMENTED** [review]
lgtm after review updates 👍

File: massa-ledger-exports/src/ledger_changes.rs
Comment (Eitu33) [inline massa-ledger-exports/src/ledger_changes.rs:827]:
```suggestion
                key.map_or(true, |k| datastore.contains_key(&k))
```

File: massa-ledger-exports/src/ledger_changes.rs
Comment (Eitu33) [inline massa-ledger-exports/src/ledger_changes.rs:831]: I get why you need that but this behaviour is very contradictory with the name of the function being `has_address_and_key`

**Comment by AurelienFT** [comment]
Thanks @Eitu33 for your reivew I have addressed it.

Just a note, we can't merge this until we have updated the project `massa-src-unit-tests`

**Review by Eitu33 — APPROVED** [review]
Sorry for the late approval I forgor ser 🌞

File: massa-async-pool/src/changes.rs
Comment (sydhds) [inline massa-async-pool/src/changes.rs:107]: Should we use an enum here or constants?

File: massa-async-pool/src/changes.rs
Comment (sydhds) [inline massa-async-pool/src/changes.rs:259]: Can we add more description about what is compensations here?

File: massa-async-pool/src/pool.rs
Comment (sydhds) [inline massa-async-pool/src/pool.rs:101]: Should be in the list as well ? (start with *)?

File: massa-execution-worker/src/interface_impl.rs
Comment (sydhds) [inline massa-execution-worker/src/interface_impl.rs:622]: We should test for some max length here to avoid allocating too much memory here?

**Review by sydhds — CHANGES_REQUESTED** [review]

File: massa-async-pool/src/changes.rs
Comment (AurelienFT) [reply massa-async-pool/src/changes.rs]: It shouldn't be here. Old copy paste.

File: massa-execution-worker/src/interface_impl.rs
Comment (AurelienFT) [reply massa-execution-worker/src/interface_impl.rs]: Good catch ! Done.

File: massa-async-pool/src/changes.rs
Comment (AurelienFT) [reply massa-async-pool/src/changes.rs]: Done.

**Pull request merged**
